front/app/modules/commercial/moderation/admin/containers/index.tsx

```diff
@@ -313,53 +316,55 @@ const Moderation = memo<Props & InjectedIntlProps>(({ className, intl }) => {
   const handleRowOnSelect = useCallback(
     (selectedModerationId: string) => {
       if (!processing) {
-        const newSelectedRows = includes(selectedRows, selectedModerationId)
-          ? selectedRows.filter((id) => id !== selectedModerationId)
-          : [...selectedRows, selectedModerationId];
-        setSelectedRows(newSelectedRows);
+        const newSelectedRows = includes(
+          selectedRowModerationIds,
+          selectedModerationId
+        )
+          ? selectedRowModerationIds.filter((id) => id !== selectedModerationId)
+          : [...selectedRowModerationIds, selectedModerationId];
+        setSelectedRowModerationIds(newSelectedRows);
       }
     },
-    [selectedRows, processing]
+    [selectedRowModerationIds, processing]
   );
 
-  const removeFlags = () => {};
-  // const removeFlags = useCallback(
-  //   async (event: React.FormEvent) => {
-  //     if (
-  //       selectedRowsWithContentWarning.length > 0 &&
-  //       !isNilOrError(moderationItems) &&
-  //       moderationStatus &&
-  //       !processing
-  //     ) {
-  //       event.preventDefault();
-  //       trackEventByName(
-  //         removeFlagButtonClicked
-  //         { selectedItemsCount: selectedRows.length }
-  //       );
-  //       setProcessing(true);
-  //       const moderations = selectedRowsWithContentWarning.map((moderationId) =>
-  //         moderationItems.find((item) => item.id === moderationId)
-  //       ) as IModerationData[];
-  //       const updatedFlaggedStatus = 'approved';
-  //       const promises = moderations.map((moderation) =>
-  //         updateModerationFlaggedStatus(
-  //           moderation.id,
-  //           moderation.attributes.moderatable_type,
-  //           updatedFlaggedStatus
-  //         )
-  //       );
-  //       await Promise.all(promises);
-  //       setProcessing(false);
-  //       setSelectedRows([]);
-  //     }
-  //   },
-  //   [selectedRows, moderationItems, moderationStatus]
-  // );
+  const removeFlags = useCallback(async () => {
```

Comment:
Only this function is really important in this file. Most of the rest is renaming things.